core/src/main/java/rdx/works/core/sargon/CryptoParametersExtensions.kt

```diff
@@ -3,25 +3,23 @@ package rdx.works.core.sargon
 import com.radixdlt.sargon.DerivationPathScheme
 import com.radixdlt.sargon.FactorSourceCryptoParameters
 import com.radixdlt.sargon.Slip10Curve
-import com.radixdlt.sargon.SupportedCurves
-import com.radixdlt.sargon.extensions.contains
-import com.radixdlt.sargon.extensions.init
+import com.radixdlt.sargon.extensions.SupportedCurves
 
 val FactorSourceCryptoParameters.Companion.babylon: FactorSourceCryptoParameters
     get() = FactorSourceCryptoParameters(
-        supportedCurves = SupportedCurves.init(Slip10Curve.CURVE25519),
+        supportedCurves = SupportedCurves(Slip10Curve.CURVE25519).asList(),
         supportedDerivationPathSchemes = listOf(DerivationPathScheme.CAP26)
     )
 
 val FactorSourceCryptoParameters.Companion.olympia: FactorSourceCryptoParameters
     get() = FactorSourceCryptoParameters(
-        supportedCurves = SupportedCurves.init(Slip10Curve.SECP256K1),
+        supportedCurves = SupportedCurves(Slip10Curve.SECP256K1).asList(),
         supportedDerivationPathSchemes = listOf(DerivationPathScheme.BIP44_OLYMPIA)
     )
 
 val FactorSourceCryptoParameters.Companion.olympiaBackwardsCompatible: FactorSourceCryptoParameters
     get() = FactorSourceCryptoParameters(
-        supportedCurves = SupportedCurves.init(Slip10Curve.SECP256K1, Slip10Curve.CURVE25519),
+        supportedCurves = SupportedCurves(Slip10Curve.SECP256K1, Slip10Curve.CURVE25519).asList(),
```

Comment:
you might wanna check for emptiness here, since UniFFI / JSON roundtrip would crash if empty.

Reply:
What do you mean, the curves are not empty in this construction.

Reply:
Protect against someone making them empty and thus crashing the app